src/main.ts

```diff
@@ -82,11 +84,15 @@ if (!innertubeClientOauthEnabled) {
         { backoffSchedule: [5_000, 15_000, 60_000, 180_000] },
         async () => {
             if (innertubeClientJobPoTokenEnabled) {
-                ({ innertubeClient, tokenMinter } = await poTokenGenerate(
-                    innertubeClient,
-                    config,
-                    innertubeClientCache,
-                ));
+                try {
+                    ({ innertubeClient, tokenMinter } = await poTokenGenerate(
+                        innertubeClient,
+                        config,
+                        innertubeClientCache,
+                    ));
+                } catch (_) {
+                    metrics?.potokenGenerationFailure.inc();
+                }
```

Comment:
this swallows the error and prevents the backoff schedule from working. You need to rethrow the error in the `catch` if you're going to wrap this. 

```suggestion
                try {
                    ({ innertubeClient, tokenMinter } = await poTokenGenerate(
                        innertubeClient,
                        config,
                        innertubeClientCache,
                    ));
                } catch (err) {
                    metrics?.potokenGenerationFailure.inc();
                    throw err;
                }
```

or alternatively

```suggestion
                  ({ innertubeClient, tokenMinter } = await poTokenGenerate(
                      innertubeClient,
                      config,
                      innertubeClientCache,
                  ).catch((err) => {
                      metrics?.potokenGenerationFailure.inc();
                      throw err;
                  }));
```

Both of these are untested, but something like this needs to happen

Reply:
Done. I added `throw new Error("xd")` to test and I get `Exception in cron handler regenerate youtube session Error: xd`. Is that the correct output?

Reply:
I'll test it out too, but that sounds right, yeah.

Reply:
Yep, this works